### ISSUE TITLE
The Flagging plugin has a bug when using MySQL 5.7

### ISSUE THREAD

**Issue by raykai** [vanilla/vanilla#3942]
if you tern the plugin on in v2.2.1 runing on MySQL 5.7 it crashes the forum dashboard.

missing the Flagging addon which has a bug in vanilla v2.2.1

![plugins - vanilla](https://cloud.githubusercontent.com/assets/8138894/15202430/28f964f2-17af-11e6-8a65-c38b085aa015.jpg)

error:

Fatal Error in PHP.trigger_error();

Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'aquaexotik0forum_vanilla.fl.ForeignID' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by

The error occurred on or near: 
/library/database/class.database.php

`408:                 // If we get here then the pdo statement prepared properly.
409:                 break;
410: 
411:             } catch (Gdn_UserException $uex) {
412:                 trigger_error($uex->getMessage(), E_USER_ERROR);
413:             } catch (Exception $ex) {
414:                 list($state, $code, $message) = $PDO->errorInfo();
415: 
416:                 // If the error code is consistent with a disconnect, attempt to retry`

Backtrace:

`/library/database/class.database.phpPHP::Gdn_ErrorHandler();
[/library/database/class.database.php:412] PHP::trigger_error();
[/library/database/class.sqldriver.php:1663] Gdn_Database->query();
[/library/database/class.sqldriver.php:614] Gdn_SQLDriver->query();
[/plugins/Flagging/class.flagging.plugin.php:34] Gdn_SQLDriver->get();
[/library/core/class.pluginmanager.php:867] FlaggingPlugin->base_getAppSettingsMenuItems_handler();
[/library/core/class.pluginmanager.php:780] Gdn_PluginManager->callEventHandler();
[/library/core/class.pluggable.php:133] Gdn_PluginManager->callEventHandlers();
[/applications/dashboard/modules/class.sidemenumodule.php:268] Gdn_Pluggable->fireEvent();
[/library/core/class.module.php:213] SideMenuModule->toString();
[/library/core/class.controller.php:1350] Gdn_Module->render();
[/applications/dashboard/views/admin.master.php:34] Gdn_Controller->renderAsset();
[/library/core/class.controller.php:1891] PHP::include();
[/library/core/class.controller.php:1318] Gdn_Controller->renderMaster();
[/library/core/class.pluggable.php:199] Gdn_Controller->xRender();
[/applications/dashboard/controllers/class.settingscontroller.php:719] Gdn_Pluggable->__call();
[/applications/dashboard/controllers/class.settingscontroller.php:719] SettingsController->render();
[/applications/dashboard/controllers/class.settingscontroller.php:719] SettingsController->plugins();
[/library/core/class.dispatcher.php:329] PHP::call_user_func_array();
[/index.php:44] Gdn_Dispatcher->dispatch();`

Variables in local scope:

`[Sql] 'select DISTINCT(fl.ForeignID) as `NumFlaggedItems`
from GDN_Flag fl
group by ForeignURL'
[InputParameters] array (
)
[Options] array (
  'Type' => 'select',
  'Slave' => NULL,
  'ReturnType' => 'DataSet',
)
[ReturnType] 'DataSet'
[tries] 2
[try] 0
[PDO] array (
)
[PDOStatement] false
[state] '42000'
[code] 1055
[message] 'Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column \'aquaexotik0forum_vanilla.fl.ForeignID\' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by'
[uex] array (
)`

![fatal error](https://cloud.githubusercontent.com/assets/8138894/15159395/97e91f76-16c2-11e6-96a5-d4ccc7b2425e.png)

**Comment by raykai** [comment]
Based on this informative blog entry, it is because some default settings in MySql have changed: http://mysqlserverteam.com/mysql-5-7-only_full_group_by-improved-recognizing-functional-dependencies-enabled-by-default/

In phpMyAdmin run two sqls:
@sql_mode

The result is a list of sql modes including ONLY_FULL_GROUP_BY. Delete this from the list and run the following sql:
SET sql_mode = 'ListWithoutTheFullGroupMode'

This will help but is a workaround and not a solution.

**Comment by alaric1** [comment]
i seam to have the same error and im using MySQL 5.7 hope to have a hot-fix for MySQL 5.7 as it comes with ubuntu 16.04

**Comment by egorsmkv** [comment]
Hello to all.

I have the same problem and created patch for resolve the problem ( based on #4020 issue and its commits ):

```patch
--- ./class.flagging.plugin.php.old	2017-11-07 23:36:35.391099375 +0200
+++ ./class.flagging.plugin.php	2017-11-07 23:36:50.427236844 +0200
@@ -28,14 +28,13 @@
      * Add Flagging to Dashboard menu.
      */
     public function base_getAppSettingsMenuItems_handler($Sender) {
-        $NumFlaggedItems = Gdn::sql()->select('fl.ForeignID', 'DISTINCT', 'NumFlaggedItems')
+        $flaggedItemsResult = Gdn::sql()->select('count(distinct(ForeignURL)) as NumFlaggedItems')
             ->from('Flag fl')
-            ->groupBy('ForeignURL')
-            ->get()->numRows();
+            ->get()->firstRow(DATASET_TYPE_ARRAY);
 
         $LinkText = t('Flagged Content');
-        if ($NumFlaggedItems) {
-            $LinkText .= ' <span class="Alert">'.$NumFlaggedItems.'</span>';
+        if ($flaggedItemsResult['NumFlaggedItems']) {
+            $LinkText .= ' <span class="Alert">'.$flaggedItemsResult['NumFlaggedItems'].'</span>';
         }
         $Menu = $Sender->EventArguments['SideMenu'];
         $Menu->AddItem('Forum', t('Forum'));
```

Also, you can download [class.flagging.plugin.php](https://raw.githubusercontent.com/vanilla/vanilla/fe439184cb7faf136b3eafbab8c6f9589bad3d5b/plugins/Flagging/class.flagging.plugin.php) file and replace with it.
